include/fiction/algorithms/simulation/sidb/determine_displacement_robustness.hpp

```diff
@@ -0,0 +1,542 @@
+                        new_pos_nw.y += static_cast<decltype(new_pos_nw.y)>(params.displacement_variations.second);
+                    }
+                    const auto all_coord = all_coordinates_in_spanned_area<cell<Lyt>>(
+                        siqad::to_siqad_coord(new_pos_se), siqad::to_siqad_coord(new_pos_nw));
+                    all_possible_sidb_misplacements.push_back(all_coord);
+                }
+                else
+                {
+                    auto new_pos_se = c;
+                    auto new_pos_nw = c;
+                    if (params.fixed_cells.find(c) == params.fixed_cells.end())
+                    {
+                        new_pos_se.x -= params.displacement_variations.first;
+                        new_pos_se.y -= params.displacement_variations.second;
+                        new_pos_nw.x += params.displacement_variations.first;
+                        new_pos_nw.y += params.displacement_variations.second;
+                    }
+
+                    const auto all_coord = all_coordinates_in_spanned_area<cell<Lyt>>(new_pos_se, new_pos_nw);
+                    all_possible_sidb_misplacements.push_back(all_coord);
+                }
+            });
+        all_possible_displacements_for_each_sidb = all_possible_sidb_misplacements;
+    }
+    /**
+     * This is a helper function, which recursively generates combinations of cell/SiDB displacements for all SiDBs
+     * based on the provided vector of displacement vectors.
+     *
+     * @param result The vector to store the generated combinations.
+     * @param current_combination The current combination being constructed.
+     * @param index The current index in the vector of displacement vectors.
+     */
+    void generate_combinations(std::vector<std::vector<cell<Lyt>>>& result, std::vector<cell<Lyt>>& current_combination,
+                               std::size_t index) noexcept
+    {
+        if (index == all_possible_displacements_for_each_sidb.size())
+        {
+            result.push_back(current_combination);
+            return;
+        }
+
+        for (const auto& c : all_possible_displacements_for_each_sidb[index])
+        {
+            current_combination.push_back(c);
+            generate_combinations(result, current_combination, index + 1);
+            current_combination.pop_back();
+        }
+    }
+    /**
+     * This function generates all SiDB layouts with displacements based on the original layout.
+     * It filters out layouts where two or more SiDBs would be on the same spot due to displacement.
+     *
+     * @return A vector containing all valid SiDB layouts with displacements.
+     */
+    [[nodiscard]] std::vector<Lyt> generate_valid_displaced_sidb_layouts()
+    {
+        std::vector<std::vector<cell<Lyt>>> all_possible_sidb_displacement{};
+        std::vector<cell<Lyt>>              current_combination{};
+        generate_combinations(all_possible_sidb_displacement, current_combination, 0);
```

Comment:
I'm a bit confused by this code. If you need to store a certain combination out of a list of possible combinations, you could use an iterator pointing to the respective combination